src/FSharpPlus/Data/NonEmptyList.fs

```diff
@@ -178,6 +184,9 @@ type NonEmptyList<'t> with
     [<EditorBrowsable(EditorBrowsableState.Never)>]
     static member Map (x: NonEmptyList<'a>, f: 'a->'b) = NonEmptyList.map f x
     
+    [<EditorBrowsable(EditorBrowsableState.Never)>]
+    static member MapIndexed (x: NonEmptyList<_>, f) = NonEmptyList.mapi f x
```

Comment:
It doesn't look like this method gets picked up by `MapIndexed.Invoke`

Reply:
means it's returning  Seq instead of NonEmptyList
![image](https://user-images.githubusercontent.com/11896817/92343219-9581bb00-f0dc-11ea-9816-77d0f41b15f0.png)

what could be the reason ?

Reply:
what is the prpose of **[<Optional>]_impl: MapIndexed** I think F# don't support implicits.
or its a trick for compiler to recognize type ?
bcs _impl is not being used anywhere in other functions as well

Reply:
I think the intent of the optional parameter in the Indexed is to be able to control the how overloading happens.

I think that issue is probably not part of this PR. @gusty has started working on allowing calls to external types https://github.com/fsprojects/FSharpPlus/pull/351

Reply:
So I'm thinking of doing a revert commit
```git revert 9e4b1d9```
since I think the PR looks good enough

Reply:
0k.